actionpack/lib/action_controller/metal/data_streaming.rb

```diff
@@ -155,7 +156,17 @@ def send_file_headers!(options)
         disposition = options.fetch(:disposition, DEFAULT_SEND_FILE_DISPOSITION)
         unless disposition.nil?
           disposition  = disposition.to_s
-          disposition += %(; filename="#{options[:filename]}") if options[:filename]
+          filename = options[:filename]
+          if filename
+            disposition += %(; filename="#{filename}")
+
+            # Support RFC 6266
+            # http://tools.ietf.org/html/rfc6266#appendix-D
+            unless filename.ascii_only?
+              encoded_filename = CGI::escape(filename)
+              disposition += %(; filename*="#{encoded_filename.encoding}''#{encoded_filename}")
```

Comment:
Yeah, no quotes. `CGI::escape` should do all the characters, but also a bunch that it shouldn't. That might not be a problem in practice, but this part is a definitely problem:

```
irb(main):004:0> CGI::escape("foo bar")
=> "foo+bar"
```

where it should use `%20`. I have a library that does this stuff in Clojure, and it a) checks that the character is specifically excluded by the spec before passing it to the escaper and b) "manually" replaces the space with the right escape sequence. You can see the relevant code here if you grok CLJ: https://github.com/Zensight/header-utils/blob/master/src/header_utils/encoding.clj#L43